### PR TITLE
Transport Legacy to Ghost and back to Legacy

### PR DESCRIPTION
**Pull request by neonphog**
## PR summary

Gives me a bind point where I can inject the multiplexer. This will mostly all be deleted once we complete the integration, but there are some lifetime fixes in here that we will keep.

## Review checklist 
- [ ] The story has unit or integration tests
- [ ] No new bugs, and any tech-debt is identified and justified
- [ ] There is enough API documentation (how to use)
- [ ] There is enough code documentation (how the code works)

### CODE REVIEW

File: crates/lib3h/src/gateway/gateway_transport.rs [hunk 1]
```diff
@@ -21,6 +21,9 @@ impl<'gateway> Transport for P2pGateway<'gateway> {
     // TODO #176 - Return a higher-level uri instead?
     fn connect(&mut self, uri: &Url) -> TransportResult<ConnectionId> {
         trace!("({}).connect() {}", self.identifier, uri);
+        if &self.identifier != "__network__" {
```
Comment (maackle) [inline crates/lib3h/src/gateway/gateway_transport.rs]: had to look up this magic string, could you import `NETWORK_GATEWAY_ID` instead?

Comment (neonphog) [reply crates/lib3h/src/gateway/gateway_transport.rs]: aw, crap... that's debugging that shouldn't be in there... we don't want to `std::process::exit(127)` ; )

Comment (neonphog) [reply crates/lib3h/src/gateway/gateway_transport.rs]: removed

**Comment by neonphog** [comment]
closing for now, will wait and impl multiplexer directly between ghost gateway and ghost engine

**Pull request closed without merging**